flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/expressions/RexNodeConverter.java

```diff
@@ -125,13 +328,127 @@ private RexNode translateScalarCall(FunctionDefinition def, List<Expression> chi
 							type.getOutputDataType().getLogicalType().copy(child.getType().isNullable())),
 					child,
 					checkOverflow);
+		} else if (BuiltInFunctionDefinitions.IN.equals(def)) {
+			Expression headExpr = children.get(1);
+			if (headExpr instanceof TableReferenceExpression) {
+				QueryOperation tableOperation = ((TableReferenceExpression) headExpr).getQueryOperation();
+				RexNode child = children.get(0).accept(this);
+				return RexSubQuery.in(
+						((FlinkRelBuilder) relBuilder).queryOperation(tableOperation).build(),
+						ImmutableList.of(child));
+			} else {
+				List<RexNode> child = convertCallChildren(children);
+				return relBuilder.call(FlinkSqlOperatorTable.IN, child);
+			}
+		} else if (BuiltInFunctionDefinitions.GET.equals(def)) {
+			RexNode child = children.get(0).accept(this);
```

Comment:
How about extracting methods like `convertGet/In/Trim`. A method of hundreds of lines is extremely hard to read.

Reply:
good suggestion.